api/adaptive_load/benchmark_result.proto

```diff
@@ -0,0 +1,41 @@
+// Protos describing the results of running a single Nighthawk benchmark and
+// scoring the resulting metrics against thresholds.
+
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "api/adaptive_load/metric_spec.proto";
+import "api/client/output.proto";
+import "google/rpc/status.proto";
+
+// Records the status of a single metric during a benchmark session.
+message MetricEvaluation {
+  // Original metric specification (plugin name, metric name).
+  MetricSpec metric_spec = 1;
+  // Numerical value of the metric measured during this benchmark session.
+  double metric_value = 2;
+  // Threshold that was used to check this metric value. Not set if the metric
+  // was only informational.
+  ThresholdSpec threshold_spec = 3;
```

Comment:
Would there be benefit in simplifying here if metrics/thresholds are named?

Reply:
The metric spec proto can be boiled down to a single string <plugin name>.<metric name>. I think we can actually just omit ThresholdSpec entirely here. It would be exactly the same verbose proto every time. From just the metric (plugin name, metric name), you can figure out the ThresholdSpec since they have a 1:1 association in the input.

If we don't include the ThresholdSpec inline in each result, we might then want to reproduce the list of (metric spec, threshold spec) at the top level in the output proto, so that the output proto can still be fully interpreted without referring back to the input.

Reply:
Went with `<plugin name>/<metric name>` because the plugin name will have dots in it.